lib/dolly/slugable.rb

```diff
@@ -0,0 +1,62 @@
+module Dolly
+  module Slugable
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    def slug_hash
+      slugable_properties.each_with_object(Hash.new) do |property, hsh|
+        hsh[property] = self.send(property)
+      end.merge(type: self.name_paramitized)
+    end
+
+    def slug
+      raise unless self.respond_to?(:slugable_properties)
+      raise if slugable_properties.none?
+      slugable_properties.map do |property|
+        self.send(property).to_s
+      end.map(&parameterize_item).join('_')
```

Comment:
should the value passed to join be configurable? it can be called from a slug_separator method that if defined thats that value and if not falls back to _

Reply:
ok, I am thinking on improve this code later, currently I am doing what other projects are doing, so I don't want to add complexity yet.

Reply:
it is now configurable